roles/satellite-clone/tasks/main.yml

```diff
@@ -166,15 +160,6 @@
       permissions. Avoid using /root.
   when: clone_foreman_dump_exists and clone_candlepin_dump_exists and (clone_no_foreman_dump_access or clone_no_candlepin_dump_access)
 
-# If there is no puppet user, the extracted config files will not have the correct puppet
-# user/group permissions with puppet 4. Installing puppetserver before extracting config_files
-# fixes this issue.
-- name: Install puppetserver to create puppet user
-  yum:
-    name: puppetserver
-    state: latest
-  when: puppet_version|int == 4
```

Comment:
Isn't this still true for all Puppetserver versions today?

Reply:
If that's true then there is a bug here, because the above task only runs `when: puppet_version|int == 4`

I wonder what is the historical reason this was needed (can check blame/logs to have a look), as I think the installer should ensure we have correct user/group/permissions as long as we reach that point

Reply:
Git forensics reveals the issue was: https://github.com/RedHatSatellite/satellite-clone/issues/328

The issue report states what was the fix, but does not state what was the original problem

It was resolved in https://github.com/RedHatSatellite/satellite-clone/commit/b0de5e6983b5ba58499cba63136b034301b96960 with a comment at https://github.com/RedHatSatellite/satellite-clone/pull/333#issuecomment-397024596 mentioning permissions issues with puppet, but not further details

In my opinion it's safe to remove this, because this code already doesn't run on any of the currently supported Satellite/Puppet versions

Reply:
Given nobody complained for a while… ;)

Reply:
I can imagine that the puppet user should be present for the backup restore to succeed, but it could very well be that the first installer run rectifies all the permissions.